src/games/strategy/engine/framework/GameRunner.java

```diff
@@ -50,14 +52,18 @@
  */
 public class GameRunner {
 
+  public enum GameMode { SWING_CLIENT, HEADLESS_BOT }
```

Comment:
That would have lead to boolean parameters, and then we have calls like this:
`setupLogging(false)`. Which is a good example of why boolean parameters are bad. There is really no good way to understand just reading that what it would mean. One tactic is to assign the `false` value to a local variable, but that is a more of a bandaid solution. Regardless, when you get in the staet that you have to jump through code to understand it, when you have to track down methods that are called to read them to understand it, it's not a good place. Imagine having to do that for 50 lines of code, tracing down a dozen methods, it would take forever to get through and to understand the code, and then in that state bugs become common since the code is often not well understood.